synapse/pipeline.yml

```
@@ -527,3 +527,39 @@ steps:
 #
 #  - label: Trigger webhook
 #    command: "curl -k https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN -d \"payload[build_num]=$BUILDKITE_BUILD_NUMBER&payload[status]=done\""
+
+  ################################################################################
+  #
+  # Complement Test Suite
+  #
+  ################################################################################
+
+  - command:
+      # Build a docker image from the checked out Synapse source
+      - "docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile ."
+      # Download Complement's source code
```

There's an argument that this incantation belongs inside the complement docker image (in the same way that sytest does it here: https://github.com/matrix-org/sytest/blob/develop/docker/bootstrap.sh#L32). But for now let's at least document what it's doing more clearly:

```suggestion
      # We use the complement:latest image to provide Complement's dependencies, but want
      # to actually run against the latest version of Complement, so download it here.
```